chainer/reporter.py

```diff
@@ -0,0 +1,325 @@
+import collections
+import contextlib
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import variable
+
+
+class Reporter(object):
+
+    """Object to which observed values are reported.
+
+    Reporter is used to collect values that users want to watch. The reporter
+    object holds a mapping from value names to the actually observed values.
```

Comment:
Related to the [above comment](https://github.com/pfnet/chainer/pull/1285/files#r69834909). It is better to add the explanation that name the object is names as `observation`.